instrument.py

```diff
@@ -96,7 +96,7 @@ def get_xdata(self):
         self.write("HORIZONTAL:RECORDLENGTH?")
         time_size = int(self.read(30))
 
-        time = numpy.arange(0,timescale*10,timescale*10/time_size)
+        time = numpy.arange(0,timescale*10,timescale*10/time_size*2) # For some reason the output was too short compared to the data buffer
```

Comment:
This may be scope-dependent, I'll look into it, but am merging for now. There is a better way to get the record length directly from the scope and that is how it should be done in an instrument-neutral way.